src/ts/address-search.ts

```diff
@@ -1,1240 +1,1441 @@
+        value: this.getWhat3WordsLookupValue(input, shouldGetSuggestions),
+      }
+    };
+
+    return JSON.stringify(data);
+  }
+
+  private getWhat3WordsLookupValue(input: string, shouldGetSuggestions: boolean): string {
+    if (input.startsWith('///') && shouldGetSuggestions) {
+      input = input.slice(3);
+    }
+
+    return input;
+  }
+
+  // Allow the keyboard to be used to either traverse up and down the picklist and select an item, or trigger a new search
+  private handleKeyboardEvent(event: KeyboardEvent): void {
+    event.preventDefault();
+
+    // Handle keyboard navigation
+    const key = this.getKey(event);
+
+    // If a picklist is populated then trigger its keyup event to select an item
+    if (this.picklist.size) {
+      if (key === 'ArrowUp' || key === 'ArrowDown' || key === 'Enter') {
+        this.picklist.keyup(event);
+        return;
+      }
+    } else {
+      // Otherwise, enable pressing 'enter' to trigger a new search
+      if (key === 'Enter') {
+        this.search(event);
+        return;
+      }
+    }
+  }
+
+  // Main function to search for an address from an input string
+  private search(event: KeyboardEvent): void {
+    event.preventDefault();
+
+    this.currentSearchTerm = this.inputs.map(input => input.value).join(',');
+
+    // Grab the country ISO code and (if it is present) the dataset name from the current value of the countryList (format: {countryIsoCode};{dataset})
+    const currentCountryInfo = this.countryCodeMapping[this.currentCountryCode] || this.currentCountryCode;
+    const countryCodeAndDataset = currentCountryInfo.split(';');
+
+    this.currentCountryCode = countryCodeAndDataset[0];
+    if (countryCodeAndDataset[1]) {
+      this.currentDataSet = countryCodeAndDataset[1];
+    }
+
+    // (Re-)set the property stating whether the search input has been reset.
+    // This is needed for instances when the search input is also an address
+    // output field. After an address has been returned, you don't want a new 
+    // search being triggered until the field has been cleared.
+    if (this.currentSearchTerm === '') {
+      this.hasSearchInputBeenReset = true;
+    }
+
+    // Check if searching is permitted
+    if (this.canSearch()) {
+      // Abort any outstanding requests
+      if (this.request.currentRequest) {
+        this.request.currentRequest.abort();
+      }
+
+      var regex = /^\/{0,}(?:[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+|[^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3}[.｡。･・︒។։။۔።।][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+([\u0020\u00A0][^0-9`~!@#$%^&*()+\-_=[{\]}\\|'<,.>?/";:£§º©®\s]+){1,3})$/;

```

Comment:
How did you get this regex ?

Reply:
what3words website https://developer.what3words.com/tutorial/detecting-if-text-is-in-the-format-of-a-3-word-address#regex